src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala

```diff
@@ -555,6 +555,25 @@ class SlickDataAccess(databaseConfig: Config) extends DataAccess {
     runTransaction(action)
   }
 
+  /** Updates the existing input symbols to replace expressions with real values **/
+  override def updateCallInputs(workflowId: WorkflowId, key: CallKey, callInputs: CallInputs): Future[Int] = {
+    type ProjectionFunction = SlickDataAccess.this.dataAccess.Symbols => (Rep[String], Rep[Option[Clob]])
+    val projectionFn: ProjectionFunction = (s: SlickDataAccess.this.dataAccess.Symbols) => (s.wdlType, s.wdlValue)
+
+    val futureCounts = callInputs map { case (inputName: String, inputValue: WdlValue) =>
+      val action = for {
+        workflowExecutionResult <- dataAccess.workflowExecutionsByWorkflowExecutionUuid(workflowId.toString).result.head
+        symbols = dataAccess.symbolsFilterByWorkflowAndScopeAndNameAndIndex(workflowExecutionResult.workflowExecutionId.get, key.scope.fullyQualifiedName, inputName, key.index.fromIndex)
+        count <- symbols.map(projectionFn).update(inputValue.wdlType.toWdlString, Option(wdlValueToDbValue(inputValue).toClob))
+      } yield count
+
+      runTransaction(action)
+    }
+
+    // Do an FP dance to get the Future[Int] from Future[Traversable[Int]]:
+    Future.sequence(futureCounts).map{_.sum}
```

Comment:
Curlies and dots don't mix with HOFs, should be `foo map {...}` or `foo.map(...)`